drivers/espi/espi_emul.c

```diff
@@ -109,7 +109,7 @@ static int espi_emul_read_lpc_request(const struct device *dev, enum lpc_periphe
 #ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
 	case EACPI_GET_SHARED_MEMORY:
 		__ASSERT_NO_MSG(api->get_acpi_shm);
-		*data = (uint32_t)api->get_acpi_shm(emul);
+		*data = (uint32_t)api->get_acpi_shm(emul->target);
```

Comment:
Fine, but here is a comment for the future.

This won't work on a 64-bit machine, right? Then account neither will the function prototype as it uses 32-bit. I suppose it is not allowed to just use 'unsigned long'?